src/promotion-builder/app/App.js

```diff
@@ -0,0 +1,187 @@
+import React, { Component } from "react";
+
+import Banner from "../common/banner/Banner";
+import Navigation from "../common/navigation/nav/Navigation";
+
+
+import AssetManager from '../features/upload-assets/main-layout/AssetManager';
+import Search from '../features/search-promotions/main-layout/Search';
+import SavePromotion from '../features/save-promotions/main-layout/SavePromotion';
+
+
+import { navigationData } from "../common/navigation/navigationData";
+
+import { promotionData } from "../../domain/promotionData";
+import { getTruthyList, arrayToMap } from "../../utils/utils";
+
+import "./app.css";
+import { VIEW } from "../../constants/constants";
+
+import bem from "bem-cn";
+const b = bem("promotion");
+
+let uuid4 = require("uuid4");
+
+class App extends Component {
```

Comment:
A lot of logic is bundled in this file. I think it may be a good idea to diagram out your app, and figure out where best some of this logic sits. Search logic should sit with Search Component and so forth. However, I know you have already started restructuring your application :)